servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/Host.java

```diff
@@ -53,17 +67,13 @@ interface Host<ResolvedAddress, C extends LoadBalancedConnection> extends Listen
     ResolvedAddress address();
 
     /**
-     * Whether the host is both considered active by service discovery and healthy by the failure
-     * detection mechanisms.
-     * @return whether the host is both active and healthy
-     */
-    boolean isActiveAndHealthy();
-
-    /**
-     * Whether the host is considered unhealthy bo the failure detection mechanisms.
-     * @return whether the host is considered unhealthy.
+     * Determine the health status of this host for selecting an appropriate connection.
+     * @param requireNewConnection whether a new connection is required.
+     * @return the {@Host.Status} that represents the anticipated ability to handle a request. The status will
+     *         consideration whether a new connection can be established if that is required, either by the pool
```

Comment:
The javadoc feels off.
Maybe simply `@return the {@Host.Status} that represents the anticipated ability to handle a request.`. I feel that the rest is an implementation detail.